api/src/test/java/org/openmrs/module/fgh/mpi/MpiContextTest.java

```diff
@@ -0,0 +1,127 @@
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ Context.class, MpiUtils.class, FhirUtils.class, MpiContext.class, BaseEventProcessor.class,
+        KeyManagerFactory.class })
+public class MpiContextTest {
+	
+	@Mock
+	private MpiHttpClient mockMpiHttpClient;
+	
+	@Mock
+	private SnapshotEventProcessor snapshotEventProcessor;
+	
+	@Mock
+	private AdministrationService adminService;
+	
+	private static final String UUID_SYSTEM = "http://test.openmrs.id/uuid";
+	
+	private static final String MESSAGE_HEADER_REFERENCE = "metadata.epts.e-saude.net/bundle";
+	
+	private static final String MESSAGE_HEADER_EVENT_URI = "urn:ihe:iti:pmir:2019:patient-feed";
+	
+	private static final AuthenticationType AUTHENTICATION_TYPE = AuthenticationType.OAUTH;
+	
+	private static final String MPI_BASE_URL = "http://sante.org.mz";
+	
+	private static final String MPI_APP_CONTENT_TYPE = "application/fhir+json";
+	
+	private static final MpiSystemType MPI_SYSTEM = MpiSystemType.SANTEMPI;
+	
+	private static final MpiSystemType MPI_SYSTEM_AS_OPENCR = MpiSystemType.SANTEMPI;
+	
+	private static final String SANTE_CLIENT_ID = "client_credentials";
+	
+	private static final String SANTE_CLIENT_SECRET = "bG6TuS3X-H1MsT4ctW!CxXjK9J4l1QpK8B0Q";
+	
+	@Mock
+	private MpiContext mpiContext;
+	
+	@Before
+	public void setup() throws Exception {
+		PowerMockito.mockStatic(Context.class);
+		PowerMockito.mockStatic(MpiUtils.class);
+		PowerMockito.mockStatic(FhirUtils.class);
+		PowerMockito.mockStatic(MpiContext.class);
+		PowerMockito.mockStatic(KeyStore.class);
+		PowerMockito.mockStatic(KeyManagerFactory.class);
+		Whitebox.setInternalState(snapshotEventProcessor, MpiHttpClient.class, mockMpiHttpClient);
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		mpiContext.setAuthenticationType(AUTHENTICATION_TYPE);
+	}
+	
+	@Test
+	public void context_shouldInitOauth() throws Exception {
```

Comment:
By the way, the OpenMRS code convention for test method names is `methodName_shouldDoXXXX()` where `methodName` is the name of the method you are testing, so this should be `init_shouldInitOauth(), this allows other developers to quickly know the method you are testing, please fix all your test method names to follow this convention